src/main/java/net/datafaker/idnumbers/IrishIdNumber.java

```diff
@@ -0,0 +1,132 @@
+ *   <li>The check character is calculated using a Modulo 23 algorithm</li>
+ *   <li>An optional suffix (A, B, H, W) may be added and affects the checksum calculation</li>
+ * </ul>
+ * <b>Reference:</b>
+ * <a href="https://www.citizensinformation.ie/en/social-welfare/irish-social-welfare-system/personal-public-service-number/#958a6b">PPSN code</a>
+ *
+ * <p>Main methods:</p>
+ * <ul>
+ *   <li>{@link #countryCode()} Returns the ISO-2 country code ("IE")</li>
+ *   <li>{@link #generateValid(BaseProviders)} Generates a valid PPSN as a string</li>
+ *   <li>{@link #generateInvalid(BaseProviders)} Generates an invalid PPSN</li>
+ *   <li>{@link #generateValid(BaseProviders, IdNumber.IdNumberRequest)} Generates a valid {@link PersonIdNumber} object</li>
+ *   <li>{@link #validateAndCheckModulo23(String)} Validates a PPSN and checks its checksum</li>
+ * </ul>
+ */
+public class IrishIdNumber implements IdNumberGenerator {
+
+    private static final Pattern IRISH_PPSN = Pattern.compile("\\d{7}[A-Z]{1,2}$");
+
+    @Override
+    public String countryCode() {
+        return "IE";
+    }
+
+    @Override
+    public String generateInvalid(final BaseProviders faker) {
+        // Generate 7 digits
+        Random random = new Random();
+        StringBuilder digits = new StringBuilder();
+        for (int i = 0; i < 7; i++) {
+            digits.append(random.nextInt(10));
+        }
+        // Append always invalid character (es: 'Z')
+        return digits.append('Z').toString();
+    }
+
+    @Override
+    public String generateValid(final BaseProviders faker) {
+        Random random = new Random();
+        int[] weights = {8, 7, 6, 5, 4, 3, 2};
+        int[] digits = new int[7];
+        String[] allowedSuffixes = {"A", "B", "H", "W"};
+        boolean addSuffix = random.nextBoolean(); // 50% chance
+        String suffix = addSuffix ? allowedSuffixes[random.nextInt(allowedSuffixes.length)] : "";
+        int sum = 0;
+
+        // Generate 7 digits
+        for (int i = 0; i < 7; i++) {
+            digits[i] = random.nextInt(10); // 0–9
+            sum += digits[i] * weights[i];
+        }
+
+        // If we have suffix include it in the checksum
+        if (!suffix.isEmpty()) {
+            char extraChar = suffix.charAt(0);
+            int extraValue = switch (extraChar) {
+                case 'A', 'B', 'H' -> extraChar - 'A' + 1;
+                case 'W' -> 0;
+                default -> 0;
+            };
+            sum += extraValue * 9;
+        }
+
+        // Calculate checksum character for modulus 23
+        int remainder = sum % 23;
+        char checkChar = (remainder == 0) ? 'W' : (char) ('A' + remainder - 1);
+
+        // Build the PPSN
+        StringBuilder ppsn = new StringBuilder();
+        for (int d : digits) {
+            ppsn.append(d);
+        }
+        ppsn.append(checkChar).append(suffix);
+
+        return ppsn.toString();
+    }
+
+    public boolean validateAndCheckModulo23(String ppsn) {
```

Comment:
I would make this method non-public (so called "package private").
Or wait, why this method is needed at all?

Reply:
I was thinking to have a separate method to divide and test the validity of the PPSN

Reply:
It's a good idea to split code into smaller methods and test them separately. 
But I assume you don't expect your users to call this method, right? @89snake89 

But then remove keyword "public" from such methods. 
1. Then it becomes "package-private", and you can still can call this method from tests.
2. But end-users cannot call this method.